### PR TITLE
Amend rec endpoint

### PR DESCRIPTION
**Pull request by fentontaylor**
### Description of Behavior

- The old endpoint /api/v1/recommendation took a JSON body with attributes `text`, `access_token`, and `user_id`
- The amended endpoint requires a 4th parameter, `genre` in the JSON body to search Spotify recommendations based on user's music preference.
- Preference allowed values should be: 'classical', 'piano', or 'electronic' (converted to 'idm' for Spotify endpoint)
- Includes tests for SpotifyService `song_params()`, updated bad request test for missing genre, and updated `app.py` test for checking that Spotify API is called with the correct parameters

### Pull request checklist
Please check if your PR fulfills the following requirements:
- [x] Tests are written
- [x] Test coverage acceptable
- [x] Sad paths accounted for
- [x] Tested in Postman 

### Merge to
- [x] Development branch
- [ ] Master

### Additional Comments

Tested in Postman to verify that tracks returned match the requested genre: PASS

### CODE REVIEW

File: tests/test_app.py [hunk 1]
```diff
@@ -113,5 +115,93 @@ def test_POST_recommendation(self, mock_text_sentiment, mock_get):
...
+            'current_mood': 0,
+            'access_token': 'totally-real-access-token',
+            'genre': 'classical'
+        }
+
+        response = self.app.post('/api/v1/recommendation',
+                                 data=json.dumps(data),
+                                 content_type='application/json')
+
+        mock_get.assert_called_with(
```
Comment (MackHalliday) [inline tests/test_app.py]: This assert_called_with test is cool!

File: services/spotify_service.py [hunk 2]
```diff
@@ -1,9 +1,10 @@
 import requests
 
 class SpotifyService:
-    def __init__(self, access_token, sentiment):
+    def __init__(self, access_token, sentiment, genre='classical'):
```
[the remark this answers is not in the document]
Comment (fentontaylor) [reply services/spotify_service.py]: Oh that's right. I initially did it this way so I wouldn't have to change all the other tests that initialized a SpotifyService. I actually meant to change that. So I'll take care of that now.

File: tests/test_app.py (hunk 1 above)
Comment (fentontaylor) [reply tests/test_app.py]: I know right! Python's mocking library is really powerful once you get the hang of it.

File: services/spotify_service.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (fentontaylor) [reply services/spotify_service.py]: Done.

**Comment by fentontaylor** [comment]
@MackHalliday Made the changes you requested. You can merge pending your approval.

**Comment by MackHalliday** [comment]
thank yoU!

**Pull request merged**